atd-vzv/src/views/summary/CrashesByTimeOfDay.js

```diff
@@ -96,13 +101,77 @@ const CrashesByTimeOfDay = () => {
       return queryUrl;
     };
 
-    // Wait for crashType to be passed up from setCrashType component,
-    // then fetch records for selected year
-    if (crashType.queryStringCrash)
-      axios.get(getFatalitiesByYearsAgoUrl()).then((res) => {
-        setHeatmapData(calculateHourBlockTotals(res.data));
+    axios.get(getFatalitiesByYearsAgoUrl()).then((res) => {
+      const formattedData = calculateHourBlockTotals(res.data);
+      setHeatmapData(formattedData);
+    });
+  }, [activeTab, crashType]);
+
+  // Query to find maximum day total per crash type
+  useEffect(() => {
+    if (maxForLegend) return;
+
+    const maxQuery = `
+    SELECT date_extract_dow(crash_date) as day, date_extract_hh(crash_date) as hour, date_extract_y(crash_date) as year, SUM(death_cnt) as death, SUM(sus_serious_injry_cnt) as serious, serious + death as all 
+    WHERE crash_date BETWEEN '${dataStartDate.format(
+      "YYYY-MM-DD"
+    )}' and '${summaryCurrentYearEndDate}' 
+    GROUP BY day, hour, year 
+    ORDER BY year 
+    |> 
+    SELECT max(death) as fatalities, max(serious) as seriousInjuries, max(all) as fatalitiesAndSeriousInjuries
+    `;
+
+    axios
+      .get(crashEndpointUrl + `?$query=` + encodeURIComponent(maxQuery))
+      .then((res) => {
+        setMaxForLegend(res.data[0]);
       });
-  }, [activeTab, crashType, hourBlockArray]);
+  }, [maxForLegend, crashType]);
+
+  // When crashType changes, add a placeholder column containing max values to weight each year consistently
+  useEffect(() => {
+    const lastRecordInHeatmapData = heatmapData[heatmapData.length - 1];
+    const isPlaceholderArraySet =
+      lastRecordInHeatmapData && lastRecordInHeatmapData.key === "";
+
+    if (!maxForLegend || heatmapData.length === 0 || isPlaceholderArraySet)
+      return;
+
+    const placeholderArray = heatmapData[0].data.map((data, i) => ({
+      key: dayOfWeekArray[i],
+      data: maxForLegend[crashType.name],
+    }));
+
+    const placeholderObjForChartWeighting = {
+      key: "",
+      data: placeholderArray,
+    };
+
+    const updatedWeightingData = [
+      ...heatmapData,
+      placeholderObjForChartWeighting,
+    ];
+
+    setHeatmapDataWithPlaceholder(updatedWeightingData);
+  }, [maxForLegend, heatmapData, crashType]);
+
+  // Hide placeholder cells
+  useEffect(() => {
+    const heatmapChildCellNumbers = [171, 172, 173, 174, 175, 176, 177];
+
+    let cellsToHide = heatmapChildCellNumbers.map((num) =>
+      document.querySelector(
+        `#demographics-heatmap > div > svg > g > g:nth-child(${num})`
+      )
+    );
+
+    cellsToHide.forEach((cell) => {
+      if (!!cell) {
+        cell.style.visibility = "hidden";
+      }
+    });
+  }, [heatmapDataWithPlaceholder, crashType, maxForLegend]);
```

Comment:
Hacking the matrix! love that document.querySelector use

Reply:
Really glad this idea panned out! Great work Mike!